### PR TITLE
Delete grouping

### PR DESCRIPTION
**Pull request by poppindouble**

### CODE REVIEW

File: client/src/http_client.rs [hunk 1]
```diff
@@ -27,6 +27,43 @@ impl ImmuxDBHttpClient {
...
+
+        match response.text() {
+            Ok(text) => Ok((status_code, text)),
+            Err(error) => Err(ImmuxDBHttpClientError::Reqwest(error.into())),
+        }
+    }
+
+    fn remove_groupings(&self, groupings: &[GroupingLabel]) -> HttpClientResult {
+        if groupings.is_empty() {
+            return Err(ImmuxDBHttpClientError::Everything);
```
Comment (blaesus) [inline client/src/http_client.rs:44]: What's `Everything`? Everything is wrong?
Also, there is nothing fundamentally wrong with passing an empty list of groupings to remove. The function just does nothing.

File: client/src/http_client.rs [hunk 2]
```diff
@@ -27,6 +27,43 @@ impl ImmuxDBHttpClient {
...
+        let url = format!("http://{}/{}", &self.host, Constants::URL_GROUPING_KEY_WORD,);
+        let groupings =
+            groupings[1..]
+                .iter()
+                .fold(format!("{}", groupings[0]), |mut acc, grouping| {
+                    acc.push_str("\r\n");
+                    let grouping_str = format!("{}", grouping);
+                    acc.push_str(&grouping_str);
+                    return acc;
+                });
```
Comment (blaesus) [inline client/src/http_client.rs:56]: Do we have to use explicit folding and concatenation?

Can we go for something like
```
        let grouping_names = groupings
            .iter()
            .map(|g| g.to_string())
            .collect::<Vec<String>>()
            .join("\r\n");

```

File: src/storage/executor/command.rs [hunk 3]
```diff
@@ -573,6 +629,24 @@ impl Command {
...
+            Command::RemoveGroupings { groupings } => {
+                let prefix = CommandPrefix::RemoveGroupings as u8;
+                let groupings_bytes: Vec<Vec<u8>> = groupings
+                    .iter()
+                    .map(|grouping| grouping.marshal())
+                    .collect();
+
+                let mut result = vec![prefix];
+
+                let num_of_groupings = groupings.len();
```
Comment (blaesus) [inline src/storage/executor/command.rs:641]: Do we need this information? Doesn't each `grouping` contain its own length?

**Review by blaesus — CHANGES_REQUESTED** [review]

File: client/src/http_client.rs (hunk 1 above)
Comment (poppindouble) [reply client/src/http_client.rs:44]: I took this `Everything` from our previous code base, gonna change it now

File: client/src/http_client.rs (hunk 2 above)
Comment (poppindouble) [reply client/src/http_client.rs:56]: good point, done

File: src/storage/executor/command.rs (hunk 3 above)
Comment (poppindouble) [reply src/storage/executor/command.rs:641]: done

**Review by blaesus — APPROVED** [review]

**Pull request merged**